blockstore.go

```diff
@@ -3,121 +3,308 @@ package lmdbbs
 import (
 	"context"
 	"fmt"
+	"math"
 	"os"
+	"sync"
 	"sync/atomic"
 
-	"github.com/bmatsuo/lmdb-go/lmdb"
 	blocks "github.com/ipfs/go-block-format"
 	"github.com/ipfs/go-cid"
 	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/ledgerwatch/lmdb-go/lmdb"
+	"github.com/pkg/errors"
 )
 
-const (
-	MaxDBs        = 1       // needs to be configurable.
-	MapSize       = 1 << 38 // 256GiB, this will need to be configurable.
-	MaxReaders    = 128
-	FreelistReuse = uint(1000) // pages, in case we decide to use https://github.com/ledgerwatch/lmdb-go (non-viral license).
+var (
+	// DefaultInitialMmapSize is the default initial mmap size to be used if the
+	// supplied value is zero or invalid. Unless modified, this value is 16MiB.
+	DefaultInitialMmapSize = int64(16 << 20) // 16MiB.
```

Comment:
that's a little small I think, 1G is pretty reasonable for a datastore.

Reply:
Ok with me. Was being conservative, but I don't see a scenario where we'd use less than 1GiB. Less demanding users of go-bs-lmdb can override the default.